torch/torch.go

```diff
@@ -6,10 +6,17 @@ package torch
 // #include "cgotorch.h"
 import "C"
 import (
-	"github.com/wangkuiyi/gotorch/aten"
+	"reflect"
 	"unsafe"
+
+	"github.com/wangkuiyi/gotorch/aten"
 )
 
+// Optimizer struct
+type Optimizer struct {
```

Comment:
It seems that we can move the newly added content into a new file `optimizer.go`.

Reply:
Yes, there will be other types of optimizers, like adam/adagrad/etc. we could move them into `optimizer.go`.